src/PoshGitTypes.ps1

```diff
@@ -283,6 +283,7 @@ class PoshGitPromptSettings {
     [PoshGitTextSpan]$DefaultPromptSuffix       = '$(">" * ($nestedPromptLevel + 1)) '
 
     [bool]$DefaultPromptAbbreviateHomeDirectory = $true
+    [bool]$DefaultPromptAbbreviateGitDirectory  = $false
```

Comment:
This is looking pretty good to me.  The only thing I wonder about is if users will understand the "interplay" between these two settings i.e. that GitDirectory effectively overrides HomeDirectory?  It's probably OK as is but if were willing to take on another breaking change, I'd almost consider an enum setting here like `$DefaultPromptPathAbbreviation` with values of `None`, `Home`, `BelowHome`, and `GitRoot`.  The deal with `Home` and `BelowHome` is that some folks want the Linux behavior that if you're in your home dir you literally see only `~`.  The current implementation was done to "cushion" the change for Windows users such that in their home dir, they see the full path `C:\Users\keith` but below that, the home dir gets swapped with `~` e.g. `~\GitHub\dahlbyk\posh-git`.

Reply:
Well the git abbreviation goal is to shorten further than home abbreviation. There shouldn't be conflicts, although you can get the git abbreviation outside the home dir (which is desirable as far as I'm concerned).

Reply:
OK, let's go with the these two settings then.

Reply:
Belated agreement with the two settings.